### PR TITLE
[mds-core] [aws-lambda] support publish

### PR DESCRIPTION
**Pull request by twelch**
Restructures packages for publish
* adds `@mds-core` scope
* adds `files` property limiting published artifacts to `dist/` folder
* removes top-level prepare script

This PR does not do anything with ladot-service-areas or the top-level types

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
       "protocol": "inspector",
       "cwd": "${workspaceFolder}/packages/mds-agency",
       "env": {
-        "npm_package_name": "mds-agency"
+        "npm_package_name": "@mds-core/mds-agency"
```
Comment (twelch) [inline .vscode/launch.json]: I assumed these need to change too

File: aws-lambda/api-gateway-authorizer/package.json [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {
   "name": "@aws-lambda/api-gateway-authorizer",
   "version": "0.1.3",
-  "private": true,
   "description": "Authorizer for API Gateway",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
+  "files": [
+    "dist/"
```
Comment (twelch) [inline aws-lambda/api-gateway-authorizer/package.json]: I need to verify this publishes what we need/expect and nothing more

Comment (twelch) [reply aws-lambda/api-gateway-authorizer/package.json]: Looks, ends up with dist/, package.json and the license

File: tsconfig.json [hunk 3]
```diff
@@ -14,25 +14,25 @@
...
-      "mds-metrics-sheet": ["packages/mds-metrics-sheet/index.ts"],
-      "mds-native": ["packages/mds-native/index.ts"],
-      "mds-policy": ["packages/mds-policy/index.ts"],
-      "mds-provider": ["packages/mds-provider/index.ts"],
-      "mds-providers": ["packages/mds-providers/index.ts"],
-      "mds-stream": ["packages/mds-stream/index.ts"],
-      "mds-test-data": ["packages/mds-test-data/index.ts"],
-      "mds-types": ["packages/mds-types/index.ts"],
-      "mds-utils": ["packages/mds-utils/index.ts"],
+      "@mds-core/mds-agency": ["packages/mds-agency/index.ts"],
```
Comment (mdurling) [inline tsconfig.json]: Now that I think about it, with the addition of the scope, these path mappings can be reduced to:
```
"@aws-lambda/*": ["aws-lambda/*"],
"@mds-core/*": ["packages/*"],
"*": ["types/*", "*"]
```

Comment (twelch) [reply tsconfig.json]: done

**Pull request merged**